assets/src/application/ui/layout/espressoTable/EspressoTable.tsx

```diff
@@ -1,24 +1,74 @@
-import React from 'react';
-import { pathOr } from 'ramda';
+import React, { useState } from 'react';
 import { Table } from 'antd';
 import { TableProps } from 'antd/lib/table/interface';
+import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
+import { insert, pathOr, pipe, remove } from 'ramda';
+
+const DragableBodyRow = (props) => {
+	return (
+		<Draggable draggableId={props['data-row-key']} index={props.index}>
+			{(provided) => (
+				<tr ref={provided.innerRef} {...provided.draggableProps} {...provided.dragHandleProps}>
+					{provided.placeholder}
+					{props.children}
+				</tr>
+			)}
+		</Draggable>
+	);
+};
 
 const EspressoTable = <RecordType extends object = any>(props: TableProps<RecordType>) => {
+	const [data, setData] = useState(props.dataSource);
+
 	// Pagination props
 	const hideOnSinglePage = pathOr(true, ['pagination', 'hideOnSinglePage'])(props);
 	const pageSize = pathOr(2, ['pagination', 'pageSize'])(props);
 	const size = pathOr('small', ['pagination', 'size'])(props);
 	const total = pathOr(null, ['pagination', 'total'])(props);
 	const pagination = { hideOnSinglePage, pageSize, size, total };
 
+	const components = {
+		body: {
+			row: DragableBodyRow,
+		},
+	};
+
+	function onDragEnd(result) {
+		const sourceRow = data[result.source.index];
+
+		if (!result.destination) {
+			return;
+		}
+
+		if (result.destination.index === result.source.index) {
+			return;
+		}
+
+		const removeSourceRow = remove(result.source.index, 1);
+		const insertSourceRowAtDestination = insert<any>(result.destination.index, sourceRow);
+		const newData = pipe(removeSourceRow, insertSourceRowAtDestination)(data);
+
+		setData(newData);
+	}
+
 	return (
-		<Table
-			columns={props.columns}
-			dataSource={props.dataSource}
-			locale={props.locale}
-			pagination={pagination}
-			size={props.size || 'middle'}
-		/>
+		<DragDropContext onDragEnd={onDragEnd}>
+			<Droppable droppableId='list'>
+				{(provided) => (
+					<div ref={provided.innerRef} {...provided.droppableProps}>
+						<Table
+							columns={props.columns}
+							className='table-drag-sorting'
+							components={components}
+							dataSource={data}
+							locale={props.locale}
+							pagination={pagination}
+							size={props.size || 'middle'}
+						/>
+					</div>
+				)}
+			</Droppable>
+		</DragDropContext>
```

Comment:
oh...hmmm... ya... does that work with the drop zone being around the entire table?